avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroInputStream.scala

```diff
@@ -15,11 +15,11 @@ trait AvroInputStream[T] {
   def tryIterator(): Iterator[Try[T]]
 }
 
-class AvroBinaryInputStream[T](in: InputStream)
+class AvroBinaryInputStream[T](in: InputStream, writerSchema: org.apache.avro.Schema = null)
```

Comment:
There is a [later commit](https://github.com/sksamuel/avro4s/pull/114/commits/960b5b3d398126e96bc923261329951c338fc227) where I update the API, 
I't let you specify the writer schema on the binary method, (not binaryEvolve).
it keeps the API nice and clean on the user site.
implicit here for the writer schema is a bit problematic because the common use case is when the same class was evolve, it now different from the time of writing.
so we can do our best by letting the user pass the actual avro schema he pulled from a schema registry or something.